cpp/example_code/s3/include/awsdoc/s3/s3_examples.h

```diff
@@ -0,0 +1,17 @@
+// Copyright Amazon.com, Inc. or its affiliates.All Rights Reserved.
+// SPDX - License - Identifier: Apache - 2.0
+
+#pragma once
+
+#include <aws/core/Aws.h>
+#include <aws/s3/model/BucketLocationConstraint.h>
+#include <awsdoc/s3/S3_EXPORTS.h>
+
```

Comment:
A minor suggestion is to use forward declaration:
```c++
namespace Aws
{
    namespace S3
    {
        class BucketLocationConstraint;
    }
}
```
Then you have don't need `#include <aws/s3/model/BucketLocationConstraint.h>`.